sourcegraph/sourcegraph
# web: storm search page implementation

## Context 

This PR has two primary purposes:
1. Start testing our approach to forking pages from the default version.
2. Start testing [react-router data loaders](https://reactrouter.com/en/main/route/loader).

## Changes 
1. Moved most of the home page logic into the Storm folder.
    - There's no markup duplication right now. The source of truth is in the Storm folder components.
    - I didn't use filename-based routing just yet. Let's review it in a separate PR.
    - Removed prop-drilling from moved components. They now rely on `useLegacyRouteContext`. Let's think about migrating those to a new approach once we have examples of the new API to access previously prop-drilled fields.
3. Created Storm routes. We can add routes to this constant one by one for new pages reimplement in Storm.
4. Moved one GraphQL used on the home page from the component into the page loader. 
    - Created page loader utils based on [the Storm research issue](https://github.com/sourcegraph/sourcegraph/issues/46470).
    - Removed changes I made to the `evaluateFeatureFlag` API for [this PR](https://github.com/sourcegraph/sourcegraph/pull/47888). Will work on them in a follow-up. 
5. Added `LegacyLayoutRouteContext` to the `WebStory` to make it compatible with updated components.
    - Added `legacyLayoutRouteContext` mock, which makes most of the `defaultProps` defined in our stories redundant.

## Test plan

1. CI
5. enable Storm in site config and open the home page
7. see the `graphql?searchQuery` in the network tab
8. all good!

## App preview:

- [Web](https://sg-web-vb-storm-2.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
